client/driver.go

```diff
@@ -58,12 +59,22 @@ func Init(config config.Config) error {
 
 	ctx, cancel := util.GetContext(context.Background())
 	defer cancel()
-
-	drv, err := driver.NewDriver(ctx, &cconfig.Driver{
-		URL:               url,
-		ApplicationId:     config.ApplicationID,
-		ApplicationSecret: config.ApplicationSecret,
-	})
+	var drv driver.Driver
+	var err error
+	if config.UseTls {
+		drv, err = driver.NewDriver(ctx, &cconfig.Driver{
```

Comment:
cfg := &cconfig.Driver{
			URL:               url,
			ApplicationId:     config.ApplicationID,
			ApplicationSecret: config.ApplicationSecret,
		}
if config.UseTLS {
   cfg.TLS = &tls.Config{}
}
drv, err = driver.NewDriver(ctx, cfg)

Reply:
simplified as you suggested.